airbyte_cdk/sources/streams/http/exceptions.py

```diff
@@ -7,14 +7,18 @@
 
 import requests
 
+from airbyte_cdk.models import FailureType
+
 
 class BaseBackoffException(requests.exceptions.HTTPError):
     def __init__(
         self,
         request: requests.PreparedRequest,
         response: Optional[Union[requests.Response, Exception]],
         error_message: str = "",
+        failure_type: Optional[FailureType] = None,
```

Comment:
I see usage of `DefaultBackoffException` in `abstract_oauth.py`. Should we update this one as well?